src/auth.rs

```diff
@@ -133,52 +201,92 @@ impl<'r> FromRequest<'r> for CreateAuthWrapper {
     type Error = anyhow::Error;
 
     async fn from_request(req: &'r Request<'_>) -> Outcome<Self, Self::Error> {
-        let (auth_data, token, config) = match extract_info(req) {
+        let (auth_data, token, config) = match extract_info(req).await {
             Ok(i) => i,
             Err(o) => return o,
         };
         // TODO remove clone, or refactor the order of validations/actions
-        match token.clone().action() {
+        match &token.action() {
             // Create actions dont have an existing orbit to authorize against, it's a node policy
             // TODO have policy config, for now just be very permissive :shrug:
             Action::Create {
                 orbit_id,
                 parameters,
                 ..
-            } => match token {
-                AuthTokens::Tezos(token_tz) => {
-                    if let Err(_) = verify_oid(orbit_id, &token_tz.pkh, parameters) {
+            } => {
+                let (method, params) = match verify_oid(&orbit_id, &parameters) {
+                    Ok(r) => r,
+                    _ => {
                         return Outcome::Failure((
                             Status::BadRequest,
                             anyhow!("Incorrect Orbit ID"),
-                        ));
+                        ))
                     }
-                    let vm = DIDURL {
-                        did: format!("did:pkh:tz:{}", &token_tz.pkh),
-                        fragment: Some("TezosMethod2021".to_string()),
-                        ..Default::default()
-                    };
-                    match create_orbit(
-                        *orbit_id,
-                        config.database.path.clone(),
-                        TezosBasicAuthorization {
-                            controllers: vec![vm],
-                        },
-                        &auth_data,
-                    )
-                    .await
-                    {
-                        Ok(Some(orbit)) => Outcome::Success(Self(orbit)),
-                        Ok(None) => {
+                };
+                let controllers = match &token {
+                    AuthTokens::Tezos(token_tz) => {
+                        match method {
+                            "tz" => {}
```

Comment:
What if I use a ZCAP signed with my Tezos wallet?

Reply:
it extracts the VM from the zcap proof